spec/lib/msf/debug_spec.rb

```diff
@@ -0,0 +1,700 @@
+        11     Command 12
+        12     Command 13
+        13     Command 14
+        14     Command 15
+        ```
+
+        </details>
+
+
+      E_LOG
+
+      expect(subject.history).to eql(error_log_output)
+    end
+
+    it "correctly retrieves & parses an empty config file & datastore" do
+      allow(::Msf::Config).to receive(:config_file).and_return(File.join(FILE_FIXTURES_PATH, 'debug', 'config_files', 'empty.ini'))
+
+      framework = double('framework')
+      expect(framework).to receive(:datastore).and_return({})
+
+
+      driver = double('driver')
+      expect(driver).to receive(:get_config_core).and_return('config_core')
+      expect(driver).to receive(:get_config).and_return({})
+      expect(driver).to receive(:get_config_group).and_return('config_group')
+      expect(driver).to receive(:active_module).and_return(nil)
+
+      expected_output = <<~OUTPUT
+      ##  %grnModule/Datastore%clr
+      The following global/module datastore, & databse setup was configured before the issue occurred:
+      <details>
+      <summary>Collapse</summary>
+
+      ```
+      The local config file is empty, no global variables are set, and there is no active module.
+      ```
+
+      </details>
+
+
+      OUTPUT
+
+      expect(subject.datastore(framework, driver)).to eql(expected_output)
+    end
+
+    it "correctly retrieves & parses a populated global datastore" do
+      allow(::Msf::Config).to receive(:config_file).and_return(File.join(FILE_FIXTURES_PATH, 'debug', 'config_files', 'empty.ini'))
+
+      framework = double('framework')
+      expect(framework).to receive(:datastore).and_return({
+                                                            'key1' => 'val1',
+                                                            'key2' => 'val2',
+                                                            'key3' => 'val3'
+                                                           })
+
+
+      driver = double('driver')
+      expect(driver).to receive(:get_config_core).and_return('group/name/1')
+      expect(driver).to receive(:get_config).and_return({})
+      expect(driver).to receive(:get_config_group).and_return('config_group')
+      expect(driver).to receive(:active_module).and_return(nil)
+
+      expected_output = <<~OUTPUT
+      ##  %grnModule/Datastore%clr
+      The following global/module datastore, & databse setup was configured before the issue occurred:
+      <details>
+      <summary>Collapse</summary>
+
+      ```
+      [group/name/1]
+      key1=val1
+      key2=val2
+      key3=val3
+      ```
+
+      </details>
+
+
+      OUTPUT
+
+      expect(subject.datastore(framework, driver)).to eql(expected_output)
+    end
+
+    it "correctly retrieves & parses a populated global datastore and current module" do
+      allow(::Msf::Config).to receive(:config_file).and_return(File.join(FILE_FIXTURES_PATH, 'debug', 'config_files', 'empty.ini'))
+
+      framework = double('framework')
```

Comment:
If we wanted to share this variable across our other `it` blocks instead of copy/pasting we can do:

```ruby
let(:framework) do
  instance_double(
    ::Msf::Framework,
    datastore: {
      'key1' => 'val1',
      'key2' => 'val2',
      'key3' => 'val3'
    }
  )
end

it "correctly retrieves and parses a populated global datastore and current module" do
   # ... Use `framework` as a normal variable ...
end
```